pmd-java/src/main/java/net/sourceforge/pmd/lang/java/typeresolution/ClassTypeResolver.java

```diff
@@ -701,6 +708,14 @@ public Object visit(ASTType node, Object data) {
     public Object visit(ASTReferenceType node, Object data) {
         super.visit(node, data);
         rollupTypeUnary(node);
+
+        JavaTypeDefinition elementTypeDef = node.getTypeDefinition();
+        if (elementTypeDef != null) {
+            // FIXME when ClassOrInterfaceType resolves type without dimensions, remove the test here
```

Comment:
ditto

Reply:
Same, this comment relates to the one you previously commented on